### ISSUE TITLE
Logits Processor Support

### ISSUE THREAD

**Issue by luikore** [antirez/ds4#142]
In vllm, there is [Logits Processor](https://docs.vllm.ai/en/latest/design/logits_processors/) API that can let CPU choose resulting token by final logits.

This is a powerful way to constraint tag completeness / reduce tool call mistakes, especially for quantized models.

SGLang was started as a framework for constraining generated tag tokens by context free grammar.

OpenAI's API has a ["lark cfg"](https://developers.openai.com/api/docs/guides/function-calling#lark-cfg) feature for function calling too.

For DS I think exposing a `logits_processor` callback will make the model a lot more playable. Some use case are:

- Perfectly parsable JSON in generated output
- Prevent listing hallucinated file names
- [Control thinking with a grammar](https://www.reddit.com/r/LocalLLaMA/comments/1sx7w55/gbnf_grammar_tweak_for_faster_qwen36_35ba3b_and/)
- Force the answer [cite from prompt](https://github.com/NVIDIA/logits-processor-zoo/blob/main/logits_processor_zoo/vllm/cite_prompt.py)

**Comment by antirez** [comment]
The reason why I'm not implementing this is because DeepSeek v4 Flash does not behave like a toy model. It follows tool calling, grammars dictated in prompts and other things verbatim, like a frontier model. I don't believe it is useful. Moreover one of the stresses of the project is that it is so simple that if you really want it, you can ask GPT to build it in a matter of minutes. But I don't really believe it is needed. Check the long context test, for instance, where a long story is narrated with persons associated with numbers in a casual way, and then the model emits perfectly the Name=34 table for all the names.
